curve25519-dalek/src/edwards.rs

```diff
@@ -183,25 +190,50 @@ impl CompressedEdwardsY {
     ///
     /// Returns `None` if the input is not the \\(y\\)-coordinate of a
     /// curve point.
-    #[rustfmt::skip] // keep alignment of explanatory comments
     pub fn decompress(&self) -> Option<EdwardsPoint> {
-        let Y = FieldElement::from_bytes(self.as_bytes());
+        let (is_valid_y_coord, X, Y, Z) = decompress::step_1(self);
+        if (!is_valid_y_coord).into() {
+            return None;
+        }
+        Some(decompress::step_2(self, X, Y, Z))
```

Comment:
This might be a bit clearer:

```suggestion
        if is_valid_y_coord.into() {
            Some(decompress::step_2(self, X, Y, Z))
        } else {
            None
        }
```